### PR TITLE
[ops, model, config, ci, docs] feat: enable Liger kernels for DeepSeek V4

### PR DESCRIPTION
**Pull request by FoolPlayer**
### What does this PR do?

> Enable real Liger dispatch for DeepSeek V4 training:
> - weighted / unweighted RMSNorm via OpSlots → Liger (and NPU unweighted fallback registration)
> - shared-expert SwiGLU keeps official `swiglu_limit` clamp, then fuses silu*mul with Liger
> - partial interleaved RoPE stays eager-only
>
> This makes the existing DeepSeek V4 config defaults (`rms_norm` / `swiglu_mlp = liger_kernel`) actually take effect through patchgen-generated modeling, while preserving `#928` eager numerical semantics when Liger is not bound.

### Checklist Before Starting

- Search for relative PRs/issues and link here: builds on DeepSeek V4 support and numerical alignment (#925, #928, #934); supersedes the unfinished Liger enablement on `codex/fix-deepseek-v4-ops-config`
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))

### Test

> - `make quality` passed
> - `pytest tests/models/test_deepseek_v4_liger_ops.py tests/models/test_deepseek_v4_fused_moe.py tests/ops/test_kernel_registry_sanity.py` → 37 passed
> - Liger numerical parity tests are CUDA + `liger_kernel` gated

### API and Usage Example

> No new public API. Existing config continues to work and now actually enables Liger for V4:
>
> ```yaml
> model:
>   ops_implementation:
>     rms_norm_implementation: liger_kernel
>     swiglu_mlp_implementation: liger_kernel
>     rotary_pos_emb_implementation: eager  # required for partial interleaved RoPE
> ```

### Design & Code Changes

> - Register Liger `rms_norm` variant `unweighted` and NPU unweighted RMSNorm fallback
> - DeepSeek V4 patchgen: add `veomni_rms_norm` / `veomni_unweighted_rms_norm` / `veomni_swiglu_mlp` OpSlots
> - Weighted RMSNorm / unweighted RMSNorm dispatch through OpSlots; eager keeps official FP32 weight multiply order
> - Shared-expert MLP always clamps with `swiglu_limit`, then optionally uses `LigerSiLUMulFunction`
> - Regenerate `patched_modeling_deepseek_v4_gpu.py` / `.diff` via patchgen
> - GPU test overrides stop forcing eager RMSNorm/SwiGLU for `deepseek_v4`
> - Docs: kernel selection + ops README note DeepSeek V4 Liger coverage
> - New unit tests for OpSlot wiring, clamp-before-Liger, and optional CUDA Liger parity

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- [x] Applied pre-commit checks (`make quality`)
- [x] Added/updated documentation
- [x] Added tests (`tests/models/test_deepseek_v4_liger_ops.py` + registry sanity)

### CODE REVIEW

File: veomni/ops/liger/__init__.py [hunk 1]
```diff
@@ -58,6 +58,17 @@ def liger_rms_norm(hidden_states, weight, eps):
...
+KERNEL_REGISTRY.register(
+    KernelSpec(
+        name="liger_kernel",
+        op_name="rms_norm",
+        variant="unweighted",
+        factory=_liger_rms_norm_factory,
+        hardware=HardwareRequirement(device_type="gpu"),
+        description="LigerKernel fused unweighted RMSNorm",
+    )
+)
```
[the remark this answers is not in the document]
Comment (FoolPlayer) [reply veomni/ops/liger/__init__.py:70]: Thanks for the review.

I checked this against our pinned `liger-kernel==0.7.0` on CUDA. `LigerRMSNormFunction` already supports the non-affine path with `W is None`:

- `ctx.elementwise_affine = W is not None`
- forward/backward both handle `W is None` (save only `X, RSTD`)

Local verification with our registered factory also works for both forward and backward when `weight=None`.

I intentionally prefer `weight=None` over a dummy `torch.ones(...)` weight:
1. It matches the true unweighted/non-affine semantics.
2. It avoids allocating and computing an unused weight gradient.

So I don't think we need a separate dummy-ones factory for the unweighted Liger variant with the current Liger version. If a future `liger-kernel` drops non-affine support, we can add a dedicated adapter then.

**Review by Luosuu — APPROVED** [review]
